code/game/objects/effects/spiderwebs.dm

```diff
@@ -101,6 +101,62 @@
 	else if(isprojectile(mover))
 		return prob(30)
 
+/obj/structure/spider/solid
+	name = "solid web"
+	icon = 'icons/effects/effects.dmi'
+	desc = "A solid thick wall of web, airtight enough to block air flow."
+	icon_state = "solidweb"
+	can_atmos_pass = ATMOS_PASS_NO
+	opacity = TRUE
+	density = TRUE
+	max_integrity = 60
+
+/obj/structure/spider/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/atmos_sensitive, mapload)
+
+/obj/structure/spider/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	if(damage_type == BURN)//the stickiness of the web mutes all attack sounds except fire damage type
+		playsound(loc, 'sound/items/welder.ogg', 100, TRUE)
+
+/obj/structure/spider/run_atom_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
+	if(damage_flag == MELEE)
+		switch(damage_type)
+			if(BURN)
+				damage_amount *= 0.75
+			if(BRUTE)
+				damage_amount *= 0.25
+	. = ..()
+
+/obj/structure/spider/passage
+	name = "web passage"
+	icon = 'icons/effects/effects.dmi'
+	desc = "messy connection of webs blocking the other side but not solid enough to prevent passage."
+	icon_state = "webpassage"
+	can_atmos_pass = ATMOS_PASS_NO
+	opacity = TRUE
+	max_integrity = 40
+	alpha = 200
+	plane = GAME_PLANE_UPPER
+	var/can_tangle = TRUE
```

Comment:
where is this used

Reply:
thats for web_passage game_plane_upper is to make it go above mobs alpha makes it somewhat transparent integrity is for structure health and can_tangle gives it the properties to be able to tagle carbon that arent spiders

Reply:
I can't find a reference to can tangle anywhere else in this code